subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/VfsRelativePath.java

```diff
@@ -89,6 +89,30 @@ public VfsRelativePath suffixStartingFrom(int startIndex) {
         return new VfsRelativePath(absolutePath, offset + startIndex);
     }
 
+    /**
+     * Returns a new relative path starting after the given prefix.
+     *
+     * E.g.
+     *   (some/path, some) -&gt; path
+     */
+    public VfsRelativePath suffixStartingFrom(VfsRelativePath prefix) {
+        return suffixStartingFrom(prefix.length() + 1);
+    }
```

Comment:
Why not flip this around, and just have two overloads of `removeAsPrefixOf()`?

Reply:
This is because we already have the method `suffixStartingFrom(int)`. So we basically have 4 cases:

| prefix type     | type of relative path | method name on VfsRelativePath |
|-----------------|-----------------------|--------------------------------|
| VfsRelativePath | String                | removeAsPrefixOf               |
| VfsRelativePath | VfsRelativePath       | suffixStartingFrom             |
| String          | VfsRelativePath       | fromChild |
| int             | VfsRelativePath       | suffixStartingFrom             |

I am not sure what good names for all those methods are. I wonder as well if we should get rid of `suffixStartingFrom(int)` and use `fromChild` instead. Looking at the callers, that should work.

Reply:
Simplifying this to 3 cases instead of 4 seems meaningful in any case. Maybe then it's easier to come up with more compatible names?

Reply:
I ended up with only two methods: `pathToChild` and `pathFromChild`.